### ISSUE TITLE
Bad error messages with an invalid GCP Key

### ISSUE THREAD

**Issue by Harwayne** [google/knative-gcp#353]
My `google-cloud-key` credential was wrong, it was for an account that no longer existed. 

I created a Broker that was backed by the PubSub Channel. It never went ready. The Channel had the error message `"Configuration "cre-default-kne-trigger-chan-publish" does not have any ready Revision."`, which it pulled verbatim from the corresponding Topic. Eventually, looking at the Pod, there was a useful error message: `{"projectId":"harwayne2","reason":"failed to verify topic exists, rpc error: code Unauthenticated desc = transport: oauth2: cannot fetch token: 400 Bad Request\nResponse: {\n  \"error\": \"invalid_grant\",\n  \"error_description\": \"Not a valid email or user ID.\"\n}"}`

The error should be propagated to at least the level of the Channel. Ideally to the Broker itself. So that just looking at the objects that I create and interact with, I know what is wrong and can fix it.

Note that the existing error message isn't great, but due to familiarity with how the system works I was able to interpret it.

**Comment by akashrv** [comment]
As part of this fix, lets also look at other errors that occur due to user input and needs to be surfaced into the K8s object status messages.

**Comment by yolocs** [comment]
Did https://github.com/google/knative-gcp/pull/341 help at all?

**Comment by Harwayne** [comment]
I am running bca49c293e5a630c8424d54d94380c88e4b0e708, which is after #341, so no I don't think that helped, or at least didn't solve it completely.

**Comment by yolocs** [comment]
Based on my test, https://github.com/google/knative-gcp/pull/384 should also fix this issue. The error message can now propagate to the topic and channel.

```
apiVersion: messaging.cloud.google.com/v1alpha1
kind: Channel
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"messaging.cloud.google.com/v1alpha1","kind":"Channel","metadata":{"annotations":{},"name":"tstchan-4","namespace":"pschan-4"},"spec":{"project":"cshou-playground","secret":{"key":"key.json","name":"google-cloud-key"}}}
  creationTimestamp: "2019-10-29T15:37:54Z"
  generation: 1
  name: tstchan-4
  namespace: pschan-4
  resourceVersion: "7626763"
  selfLink: /apis/messaging.cloud.google.com/v1alpha1/namespaces/pschan-4/channels/tstchan-4
  uid: 1356cf8d-fa62-11e9-b004-42010a8a0065
spec:
  project: cshou-playground
  secret:
    key: key.json
    name: google-cloud-key
status:
  address:
    url: http://cre-tstchan-4-chan-publish.pschan-4.svc.cluster.local
  conditions:
  - lastTransitionTime: "2019-10-29T15:38:01Z"
    status: "True"
    type: Addressable
  - lastTransitionTime: "2019-10-29T15:38:48Z"
    message: 'Failed to create Topic: "failed to verify topic exists, rpc error: code
      = Unauthenticated desc = transport: oauth2: cannot fetch token: 400 Bad Request\nResponse:
      {\n  \"error\": \"invalid_grant\",\n  \"error_description\": \"Invalid JWT Signature.\"\n}"'
    reason: CreateFailed
    status: "False"
    type: Ready
  - lastTransitionTime: "2019-10-29T15:38:48Z"
    message: 'Failed to create Topic: "failed to verify topic exists, rpc error: code
      = Unauthenticated desc = transport: oauth2: cannot fetch token: 400 Bad Request\nResponse:
      {\n  \"error\": \"invalid_grant\",\n  \"error_description\": \"Invalid JWT Signature.\"\n}"'
    reason: CreateFailed
    status: "False"
    type: TopicReady
  observedGeneration: 1
  subscribableStatus: {}
  subscribablestatus: {}
  topicId: cre-chan-1356cf8d-fa62-11e9-b004-42010a8a0065
```

**Comment by nachocano** [comment]
Fixed by https://github.com/google/knative-gcp/pull/384

**Comment by Harwayne** [comment]
/close

Closed by #384.
